lib/resty/rediscluster.lua

```diff
@@ -137,6 +189,157 @@ local function split(s, delimiter)
 end
 
 
+-- cache only IPs
+local function parse_masters(self, redis_client)
+    if not redis_client then
+        return nil, "redis_client can not be nil"
+    end
+
+    local nodes, err = redis_client:cluster("nodes")
+    if not nodes then
+        return nil, "failed to fetch nodes: " .. err
+    end
+
+    -- cahce only master nodes
+    local master_list = {}
+
+    -- bf7d28ccd79ca000547b8d6dfc042ca08b5d8b1a 172.26.0.25:6379@16379 slave 86bc3b462782f48ee2b56939107744a792dc3575 0 1718355186000 1 connected
+    -- 79f6be094052b48c63967bc494e493a739016305 172.26.0.26:6379@16379 slave 438d9bc4c865e342a0a6dd1b560ac5d3f0c92662 0 1718355186000 2 connected
+    -- 72d111446f11ee900df33f8a91425bb6533be0b6 172.26.0.23:6379@16379 master - 0 1718355187000 3 connected 10923-16383
+    -- 438d9bc4c865e342a0a6dd1b560ac5d3f0c92662 172.26.0.22:6379@16379 master - 0 1718355186902 2 connected 5461-10922
+    -- 0e0da2991a4f713691c209ba2b7467d2a8223e9c 172.26.0.24:6379@16379 slave 72d111446f11ee900df33f8a91425bb6533be0b6 0 1718355187920 3 connected
+    -- 86bc3b462782f48ee2b56939107744a792dc3575 172.26.0.21:6379@16379 myself,master - 0 1718355182000 1 connected 0-5460
+    local node_lst = split(nodes, char(10))  -- char(10) is "\n"
```

Comment:
Good point. Let's use `_list` instead of `_lst`.

---

I found `string.char` is used in other place, so I use the same style. I will add a new commit to use the char directly instead of ASCII code.